Source/JavaScriptCore/wasm/WasmB3IRGenerator.cpp

```diff
@@ -2349,44 +2358,75 @@ auto B3IRGenerator::addArrayNew(uint32_t typeIndex, ExpressionType size, Express
         initValue = patchpoint;
     }
 
-    // FIXME: Emit this inline.
-    // https://bugs.webkit.org/show_bug.cgi?id=245405
-    result = push(m_currentBlock->appendNew<CCallValue>(m_proc, toB3Type(Types::Arrayref), origin(),
-        m_currentBlock->appendNew<ConstPtrValue>(m_proc, origin(), tagCFunction<OperationPtrTag>(operationWasmArrayNew)),
-        instanceValue(), m_currentBlock->appendNew<Const32Value>(m_proc, origin(), typeIndex),
-        get(size), initValue));
+    result = pushArrayNew(typeIndex, initValue, size);
 
     return { };
 }
 
 auto B3IRGenerator::addArrayNewDefault(uint32_t typeIndex, ExpressionType size, ExpressionType& result) -> PartialResult
 {
-    // FIXME: Abstract and generalize with addArrayNew.
+#if ASSERT_ENABLED
     Wasm::TypeDefinition& arraySignature = m_info.typeSignatures[typeIndex];
     ASSERT(arraySignature.is<ArrayType>());
-    Wasm::Type elementType = arraySignature.as<ArrayType>()->elementType().type;
+#endif
 
     Value* initValue;
-    if (Wasm::isRefType(elementType))
+    if (Wasm::isRefType(arraySignature.as<ArrayType>()->elementType().type))
         initValue = m_currentBlock->appendNew<Const64Value>(m_proc, origin(), JSValue::encode(jsNull()));
     else
         initValue = m_currentBlock->appendNew<Const64Value>(m_proc, origin(), 0);
 
+    result = pushArrayNew(typeIndex, initValue, size);
+
+    return { };
+}
+
+auto B3IRGenerator::addArraySet(uint32_t typeIndex, ExpressionType arrayref, ExpressionType index, ExpressionType value) -> PartialResult
```

Comment:
Minor nit: could we move this below `addArrayGet`? (just to keep with the declaration order, and reduce the git diff)